import-export-cli/impl/logger.go

```diff
@@ -204,3 +233,128 @@ func SetAPILoggingLevel(credential credentials.Credential, environment, apiId, t
+	}
+
+	formatContext := formatter.NewContext(os.Stdout, format)
+
+	renderer := func (w io.Writer, t *template.Template) error {
+		for _, component := range components {
+			if err := t.Execute(w, newLoggingCorrelationComponentFromComponent(component)); err != nil {
+				return err
+			}
+			_, _ = w.Write([]byte{'\n'})
+		}
+		return nil
+	}
+	
+	correlationTableHeaders := map[string]string{
+		"Name": "COMPONENT_NAME", 
+		"Enabled": "ENABLED",
+		"Properties": "PROPERTIES",
+	}
+
+	if err := formatContext.Write(renderer, correlationTableHeaders); err != nil {
+		fmt.Println("Error executing template", err.Error())
+	}
+}
+func SetCorrelationLoggingComponent(credential credentials.Credential, environment, componentName, enabled, deniedThreads string) (*resty.Response, error) {
+	// Base64 encoding the credentials
+	b64encodedCredentials := credentials.GetBasicAuth(credential)
+
+	correlationDevOpsEP := utils.GetCorrelationLoggingEndPointOfEnv(environment, utils.MainConfigFilePath)
+
+	// Prepping the headers
+	headers := make(map[string]string)
+	headers[utils.HeaderAuthorization] = utils.HeaderValueAuthBasicPrefix + " " + b64encodedCredentials
+	headers[utils.HeaderAccept] = utils.HeaderValueApplicationJSON
+
+	resp, err := utils.InvokeGETRequest(correlationDevOpsEP, headers)
+
+	if err != nil {
+		utils.HandleErrorAndExit("Unable to connect to " + correlationDevOpsEP, err)
+	}
+
+	utils.Logln(utils.LogPrefixInfo + "GET Response : " + resp.Status())
+
+	if resp.StatusCode() != http.StatusOK {
+		return nil, errors.New(string(resp.Body()))
+	}
+
+	correlationComponentsResponse := &utils.CorrelationComponentList{}
+	unmarshalError := json.Unmarshal([]byte(resp.Body()), &correlationComponentsResponse)
+
+	if unmarshalError != nil {
+		utils.HandleErrorAndExit(utils.LogPrefixError + "invalid JSON response", unmarshalError)
+	}
+
+	responseComponents := correlationComponentsResponse.Components
+	requestComponents  := make([]utils.CorrelationComponent,0)
+
+	for  _ , cc := range  responseComponents{
+		if cc.Name == componentName {
+			cc.Enabled = enabled
+			if(len(cc.Properties)>0){
+				if(cc.Properties[len(cc.Properties)-1].Name == "deniedThreads"){
+					cc.Properties[len(cc.Properties)-1].Value = strings.Split(deniedThreads, ",")
+				}
+			}
+		}
+		requestComponents = append(requestComponents, cc)
+	}
+	
+	b,err := json.Marshal(requestComponents)
+	if err != nil {
+		utils.Logln("Error when creating a json ")		
+		return nil, nil 
+	}
+	
+	headers[utils.HeaderContentType] = utils.HeaderValueApplicationJSON
+	body := string(b)
+	body = "{\"components\":" + body + "}"
+	putResp, err := utils.InvokePutRequest(nil, correlationDevOpsEP, headers, body)
+
+	if err != nil {
+		utils.HandleErrorAndExit("Unable to connect to "+correlationDevOpsEP, err)
+	}
+
+	utils.Logln(utils.LogPrefixInfo+"PUT Response:", putResp.Status())
+	if putResp.StatusCode() == http.StatusOK {
+		return putResp, nil
+	}
+	return nil, errors.New(string(putResp.Body()))
+}
```

Comment:
Shall we add a newline at the EOF?

Reply:
Fixed via https://github.com/wso2/product-apim-tooling/commit/5584c0cda4fcd40ca92ceb0a8368f41672eba295